sim/battle.js

```diff
@@ -1715,33 +1998,49 @@ class Battle extends Dex.ModdedDex {
 
 		this.event.modifier = ((previousMod * nextMod + 2048) >> 12) / 4096;
 	}
+
+	/**
+	 * @param {number} value
+	 * @param {number | number[]} numerator
+	 * @param {number} [denominator]
+	 */
 	modify(value, numerator, denominator) {
 		// You can also use:
 		// modify(value, [numerator, denominator])
 		// modify(value, fraction) - assuming you trust JavaScript's floating-point handler
 		if (!denominator) denominator = 1;
-		if (numerator && numerator.length) {
+		if (Array.isArray(numerator)) {
 			denominator = numerator[1];
 			numerator = numerator[0];
 		}
 		let modifier = Math.floor(numerator * 4096 / denominator);
 		return Math.floor((value * modifier + 2048 - 1) / 4096);
 	}
+
+	/**
+	 * @param {string | Move} move
+	 */
 	getCategory(move) {
 		move = this.getMove(move);
 		return move.category || 'Physical';
 	}
+
+	/**
+	 * @param {Pokemon} pokemon
+	 * @param {Pokemon} target
+	 * @param {string | number | Move} move
+	 * @param {boolean} [suppressMessages]
+	 */
 	getDamage(pokemon, target, move, suppressMessages) {
 		if (typeof move === 'string') move = this.getMove(move);
 
 		if (typeof move === 'number') {
-			move = {
-				basePower: move,
-				type: '???',
-				category: 'Physical',
-				willCrit: false,
-				flags: {},
-			};
+			let basePower = move;
+			move = this.getMove('');
```

Comment:
`getMove` does not return a mutable move. :|

You may be thinking of `getMoveCopy` which does.

Reply:
`new Data.Move` may be more straightforward?

Reply:
Whoops; I changed it to <code>new Data.Move</code>.